jerry-core/lit/lit-char-helpers.c

```diff
@@ -14,9 +14,89 @@
  */
 
 #include "lit-char-helpers.h"
-
+#include "lit/lit-unicode-ranges.inc.h"
 #include "lit-strings.h"
 
+#define NUM_OF_ELEMENTS(array) (sizeof (array) / sizeof ((array)[0]))
+
+/**
+ * Binary search algorithm that searches the a
+ * charachter in the given char array.
+ *
+ * @return true - if the charachter is in the given array
+ *         false - otherwise
+ */
+static bool
+search_char_in_char_array (ecma_char_t c,         /**< code unit */
+                           ecma_char_t *array,    /**< array */
+                           size_t size_of_array)  /**< length of the array */
+{
+  int bottom = 0;
+  int top = (int) size_of_array - 1;
+
+  while (bottom <= top)
+  {
+    int middle = (bottom + top) / 2;
+    ecma_char_t current = array[middle];
+
+    if (current == c)
+    {
+      return true;
+    }
+
+    if (c < current)
+    {
+      top = middle-1;
+    }
+    else
+    {
+      bottom = middle+1;
+    }
+  }
+
+  return false;
+} /* search_char_in_char_array */
+
+/**
+ * Binary search algorithm that searches a charachter in the given intervals.
+ * Intervals specifed by two arrays. The first one contains the starting points
+ * of the intervals, the second one contains the length of them.
+ *
+ * @return true - if the the charachter is included (inclusively) in one of the intervals in the given array
+ *         false - otherwise
+ */
+static bool
+search_char_in_interval_array (ecma_char_t c,             /**< code unit */
+                               ecma_char_t *array_sp,     /**< array of interval starting points */
+                               uint8_t *lengths,          /**< array of interval lengths */
+                               size_t size_of_array)      /**< length of the array */
+{
+  int bottom = 0;
+  int top = (int) size_of_array - 1;
```

Comment:
I think size_of_array should be int, or top and bottom as size_t. There is no point in changing the sign.